app/src/main/java/com/toandv/demo/musicplayer/services/PlayerService.kt

```diff
@@ -0,0 +1,220 @@
+package com.toandv.demo.musicplayer.services
+
+import android.annotation.SuppressLint
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Intent
+import android.net.Uri
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import androidx.media.app.NotificationCompat.MediaStyle
+import com.toandv.demo.musicplayer.R
+import com.toandv.demo.musicplayer.data.Song
+import com.toandv.demo.musicplayer.ui.MainActivity
+import com.toandv.demo.musicplayer.utils.Actions
+import com.toandv.demo.musicplayer.utils.Extras
+import com.toandv.demo.musicplayer.utils.ImageUtils
+import com.toandv.demo.musicplayer.utils.Noti
+
+class PlayerService : Service() {
+
+    private val binder: LocalBinder = LocalBinder()
+    private lateinit var musicPlayer: MusicPlayer
+    private lateinit var notificationManager: NotificationManagerCompat
+    private lateinit var builder: NotificationCompat.Builder
+    private lateinit var actionPlay: NotificationCompat.Action
+    private lateinit var actionPause: NotificationCompat.Action
```

Comment:
Nếu không inject từ bên ngoài vào thì khai báo nullable thôi ông ơi 
```suggestion
    private var musicPlayer: MusicPlayer?
    private var notificationManager: NotificationManagerCompat?
    private var builder: NotificationCompat.Builder?
    private var actionPlay: NotificationCompat.Action?
    private var actionPause: NotificationCompat.Action?
```
`Normally, properties declared as having a non-null type must be initialized in the constructor. However, fairly often this is not convenient. For example, properties can be initialized through dependency injection, or in the setup method of a unit test. In this case, you cannot supply a non-null initializer in the constructor, but you still want to avoid null checks when referencing the property inside the body of a class.
`
https://viblo.asia/p/co-nen-su-dung-kotlin-lateinit-var-hay-khong-LzD5dbneZjY